gcloud/connection.py

```diff
@@ -45,6 +39,11 @@ def __init__(self, credentials=None):
         self._http = None
         self._credentials = credentials
 
+    @gcloud._UserAgentReifyProperty  # pragma: NO COVER
```

Comment:
That would require modifying to always use `user_agent` on the instance that owns the property instead of using `wrapped.__name__`. WDYT of that?